pkgs/by-name/pi/pihole/package.nix

```diff
@@ -0,0 +1,257 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, makeWrapper
+, installShellFiles
+, bash
+, coreutils
+, curl
+, dig
+, gawk
+, glibc
+, gnugrep
+, gnused
+, iproute2
+, killall
+, netcat
+, nettools
+, pihole-ftl
+, procps
+, ncurses
+, util-linux
+, resholve
+, sqlite
+, libidn2
+, systemd
+, stateDir ? "/etc/pihole"
+, ...
+}:
+
+(resholve.mkDerivation (finalAttrs: {
```

Comment:
I think `resholve`'s `mkDerivation` might not support `finalAttrs` yet; hence, "value is a function while a set was expected".

Reply:
Apologies for the extra work. 

I'm not exactly sure what the best course of action is on resholve's end, but after reading around a bit it seems like a lot of the PRs for supporting this in more popular builders have bogged down, including a more general implementation that ~smells like what I'll need to update resholve to use. 

I went ahead and opened an issue to track this on my end:
- abathur/resholve#107